### DISCUSSION TITLE
Checkbox Group

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#223] [Component suggestions]
<img width="480" alt="visuele weergave van het checkbox group component" src="https://github.com/nl-design-system/backlog/assets/545207/c6712fa8-8333-428c-8c05-c6b73a6a41f1">

We willen graag meer leren over de Checkbox Group component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1708952404215789)

---

## Naam
Checkbox Group

## Doel
Biedt de mogelijkheid om geen, één of meerdere opties te selecteren uit een lijst.

## Gerelateerde componenten
[Checkbox](https://github.com/orgs/nl-design-system/discussions/306), [Fieldset](https://github.com/orgs/nl-design-system/discussions/309), [Radio Group](https://github.com/orgs/nl-design-system/discussions/222)

## Onderzoek

[Checkboxes - GOV.UK](https://design-system.service.gov.uk/components/checkboxes/)
Onder 'Research on this component' worden bevindingen genoteerd. Er is ook een [link naar de Checkboxes GitHub Discussion](https://github.com/alphagov/govuk-design-system-backlog/issues/37) waar meer informatie, onderzoek en aanpassingen worden gedeeld.

In een [comment bij Checkbox Github Discussion](https://github.com/alphagov/govuk-design-system-backlog/issues/37#issuecomment-867539860) wordt meer tekst en uitleg gegeven wat GOV.UK heeft gedaan rondom het toevoegen van een 'of' divider om een optie zoals 'Geen van bovenstaande' apart van de andere opties te presenteren. Update: inmiddels is er ook een [artikel de 'None' checkbox](https://designnotes.blog.gov.uk/2021/11/15/letting-users-tick-a-none-checkbox/) geschreven.

In een [comment bij Checkbox Github Discussion](https://github.com/alphagov/govuk-design-system-backlog/issues/37#issuecomment-871513014) wordt meer tekst en uitleg gegeven wat GOV.UK heeft gedaan rondom het conditioneel onthullen van een text input in combinatie met een Checkbox Group.

[Myth #23: Choices should always be limited to 7+/-2 - UX Myths](https://uxmyths.com/post/931925744/myth-23-choices-should-always-be-limited-to-seven)
Artikel beschrijft de mythe rondom het aanhouden van maximaal 7 opties.

**Comment by sergei-maertens** [comment]
## Open Formulieren

[Storybook](https://open-formulieren.github.io/open-forms-sdk/?path=/story/form-io-components-vanilla-selectboxes--default)

---

### Screenshots / Varianten

**Storybook**
Default
![Scherm­afbeelding 2024-02-28 om 11 02 59](https://github.com/nl-design-system/backlog/assets/545207/182efb1f-009d-4a7a-b43a-523cf015d563)

---

### Notities
- Ik merk nu dat de styling behoorlijk vernaggeld is :thinking:

**Comment by Rozerinay** [comment]
Ik heb één inzicht gevonden uit ons onderzoek naar formulieren in 2022.

De checkboxes die we hadden ontworpen waren 44px groot. Gebruikers waardeerden deze grootte, omdat het eenvoudig te selecteren was op touchscreen devices.

Het is dus belangrijk om rekening te houden met het bieden van voldoende ruimte bij checkboxes en/of tussen form field checkbox options.

<img width="468" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/e9cacce0-7f8e-46e8-a502-270714c5b424">

**Comment by jeffreylauwers** [comment]
## Amsterdam

- [Storybook (Checkbox)](https://designsystem.amsterdam/?path=/docs/components-forms-checkbox--docs)
- [GitHub](https://github.com/Amsterdam/design-system/tree/develop/packages/css/src/components/checkbox)
- [Figma](https://www.figma.com/file/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?type=design&node-id=9600-33863&mode=design&t=p24VVZsX9wzPsnkW-4)

---

### Screenshots / Varianten

**Storybook**

Checkbox group

![Scherm­afbeelding 2024-09-13 om 14 42 55](https://github.com/user-attachments/assets/2ef789b2-c03c-4d25-8977-6b14d7e31cdc)

Checkbox group with error

![Scherm­afbeelding 2024-09-13 om 14 43 01](https://github.com/user-attachments/assets/58ef89c5-4b30-4094-a6de-46a84545ba6e)


**Figma**
Form field met Checkbox group
![Scherm­afbeelding 2024-02-28 om 11 05 21](https://github.com/nl-design-system/backlog/assets/545207/1d96f115-d3a6-4b41-b084-2097b8c49720)

Form field met Checkbox group - validatie
![Scherm­afbeelding 2024-02-28 om 11 05 31](https://github.com/nl-design-system/backlog/assets/545207/8635bcb4-f9f5-48b0-a7c6-a2c87ab9fa69)

---

### Notities
- Documentatie beschikbaar (Storybook)
- Bekijk ook de [GitHub Discussion - Checkbox toevoeging van Amsterdam](https://github.com/orgs/nl-design-system/discussions/306#discussioncomment-10794988).

**Comment by jeffreylauwers** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/css-input-form-field--docs#checkbox-group)
- [Figma - Huidig](https://www.figma.com/file/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?type=design&node-id=1450-4570&mode=design&t=I9joNzh470AEai4d-4)
- [Figma - WIP](https://www.figma.com/file/x4RkF6BIdrNZbh7D53NTzB/%F0%9F%92%A0-NLDS---Den-Haag---Bibliotheek?type=design&node-id=954-2227&mode=design&t=XLCMcvNvlWi60dcQ-4)

---

### Screenshots / Varianten

**Storybook**

Checkbox Group
![Scherm­afbeelding 2024-02-28 om 11 19 49](https://github.com/nl-design-system/backlog/assets/545207/abcb5d50-820b-4730-bd8a-88bebb18e175)

Checkbox Group With Description
![Scherm­afbeelding 2024-02-28 om 11 19 53](https://github.com/nl-design-system/backlog/assets/545207/f2ac43fe-eeb8-4eb2-9c72-8eb7cc1c15b6)


**Figma - Huidig**

Checkbox group
![Scherm­afbeelding 2024-02-28 om 11 22 13](https://github.com/nl-design-system/backlog/assets/545207/9bde25d8-8405-420d-ab2e-c5cfe07cb47b)


**Figma - WIP**

Checkbox group
![Scherm­afbeelding 2024-02-28 om 11 24 22](https://github.com/nl-design-system/backlog/assets/545207/f624732b-14e9-41d4-ae40-b64a0b0f17cc)

Mogelijkheid voor description per optie.
![Scherm­afbeelding 2024-02-28 om 11 29 02](https://github.com/nl-design-system/backlog/assets/545207/cd86c060-9f2b-45e2-aee0-ad6a2bbb8caf)

---

### Notities
- [Zie comment van Rozerin](https://github.com/orgs/nl-design-system/discussions/223#discussioncomment-8593297) over onderzoek waarbij grotere checkboxes werden ingezet.
- Figma - Huidig component heeft ook een mogelijkheid voor 'Large' variant. Hierbij schaalt het label mee.
- Figma - WIP component heeft eenzelfde description opzet als bij [radio group](https://github.com/orgs/nl-design-system/discussions/222#discussioncomment-8605711). 
- Figma - WIP component heeft eenzelfde validatie opzet als bij [radio group](https://github.com/orgs/nl-design-system/discussions/222#discussioncomment-8605711). 
- Bekijk ook de [GitHub Discussion - Checkbox toevoeging van Den Haag](https://github.com/orgs/nl-design-system/discussions/306#discussioncomment-10796540).

**Comment by jeffreylauwers** [comment]
## DSO Toolkit

[Storybook](https://storybook.dso-toolkit.nl/62.8.4/?path=/story/html-css-form-groups-radios--radios&viewMode=story)

---

### Storybook

Groups / checkboxes - checked
![Scherm­afbeelding 2024-02-28 om 11 32 18](https://github.com/nl-design-system/backlog/assets/545207/995db843-29ec-4afc-a945-055162348110)

Groups / checkboxes - validatie en helptext
![Scherm­afbeelding 2024-02-28 om 11 33 23](https://github.com/nl-design-system/backlog/assets/545207/3dc71e4e-4f45-4239-abc2-7932aca06e90)

Nested combinatie met checkboxes.
![Scherm­afbeelding 2024-02-27 om 15 14 36](https://github.com/nl-design-system/backlog/assets/545207/0a6bb294-cd8f-47d5-beb1-efd42bacaaf2)


---

### Notities
- Optie voor nested weergave.
- Bekijk ook de [GitHub Discussion - Checkbox toevoeging van DSO](https://github.com/orgs/nl-design-system/discussions/306#discussioncomment-10796787).

**Comment by jeffreylauwers** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/checkbox)

---

### Screenshots / Varianten

**Documentatie website**

Default
![Scherm­afbeelding 2024-02-28 om 11 37 33](https://github.com/nl-design-system/backlog/assets/545207/d06f0294-c0dc-46d9-8992-dae3f17f8f35)


Horizontale checkbox in verticaal form
Wanneer je maximaal 2 checkboxen hebt met een kort label, mag je deze om ruimte te besparen in een verticaal form naast elkaar zetten.
![Scherm­afbeelding 2024-02-28 om 11 39 39](https://github.com/nl-design-system/backlog/assets/545207/9e64766f-02ce-4713-8acf-fb021282921a)

Horizontale checkbox in horizontaal form
In een horizontaal form is het mogelijk meerdere checkboxen naast elkaar te weergeven. Aangeraden wordt om niet meer dan 3 checkboxen te gebruiken. Horizontaal gebruik van formulier elementen is alleen bedoeld voor medewerkersschermen.
![Scherm­afbeelding 2024-02-28 om 11 39 44](https://github.com/nl-design-system/backlog/assets/545207/87b03c13-6315-4112-85b4-fc3af49937e5)

Geneste checkbox
Met geneste checkboxen kun je eenvoudig een hele groep checkboxen aan of uit zetten.
![Scherm­afbeelding 2024-02-28 om 11 37 59](https://github.com/nl-design-system/backlog/assets/545207/432d4486-a4b6-4bae-a4d9-26ced8442bdc)

Validatie
![Scherm­afbeelding 2024-02-28 om 11 38 19](https://github.com/nl-design-system/backlog/assets/545207/6099a730-5b22-4662-8dce-86ab8e06b65e)

---


### Notities
- Mogelijkheid beschikbaar om opties horizontaal te tonen.
- Mogelijkheid beschikbaar om opties te nesten.
- Bekijk ook de [GitHub Discussion - Checkbox toevoeging van DUO](https://github.com/orgs/nl-design-system/discussions/306#discussioncomment-10796863).

**Comment by jeffreylauwers** [comment]
## OpenGemeenten
[Patternlab](https://patternlab.lenteveld.nl/index.html)

---

### Screenshots / Varianten

**Patternlab**

[MultiCheckbox](https://patternlab.lenteveld.nl/patterns/01-molecules-080-Forms-And-Search-MultiCheckbox-MultiCheckbox/01-molecules-080-Forms-And-Search-MultiCheckbox-MultiCheckbox.html)
![Scherm­afbeelding 2024-02-28 om 11 51 51](https://github.com/nl-design-system/backlog/assets/545207/6052c132-321a-45b5-ba94-89a647b1b446)

[MultiCheckbox Description](https://patternlab.lenteveld.nl/patterns/01-molecules-080-Forms-And-Search-MultiCheckbox-MultiCheckbox-Description/01-molecules-080-Forms-And-Search-MultiCheckbox-MultiCheckbox-Description.html)
![Scherm­afbeelding 2024-02-28 om 11 51 55](https://github.com/nl-design-system/backlog/assets/545207/15a95f6d-7d9f-4da1-8f56-50cae3ea71d4)

[MultiCheckbox Error](https://patternlab.lenteveld.nl/patterns/01-molecules-080-Forms-And-Search-MultiCheckbox-MultiCheckbox-Error/01-molecules-080-Forms-And-Search-MultiCheckbox-MultiCheckbox-Error.html)
![Scherm­afbeelding 2024-02-28 om 11 52 01](https://github.com/nl-design-system/backlog/assets/545207/bd88e205-e0f6-4f56-a32f-b008f3028255)

---

### Notities
Bekijk ook de [GitHub Discussion - Checkbox toevoeging van OpenGemeenten](https://github.com/orgs/nl-design-system/discussions/306#discussioncomment-10797703).

**Comment by jeffreylauwers** [comment]
## Rotterdam

[Figma](https://www.figma.com/file/ZWSC4gCrOXRUR9UX3aoZ8x/Gemeente-Rotterdam-Design-System?type=design&node-id=521-28182&mode=design&t=B6BdewKoEAslWiOU-4)

---

### Screenshots / Varianten

**Figma**

Groep inclusief validatie
![Scherm­afbeelding 2024-02-28 om 11 57 40](https://github.com/nl-design-system/backlog/assets/545207/1f832b3d-988d-40e2-8acd-98808f737847)


---

### Notities
- Onder de noemer: Toggles.
- Bekijk ook de [GitHub Discussion - Checkbox toevoeging van Rotterdam](https://github.com/orgs/nl-design-system/discussions/306#discussioncomment-10797825).

**Comment by jeffreylauwers** [comment]
## RVO

- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-checkbox-group--documentatie)
- [GitHub](https://github.com/nl-design-system/rvo/tree/master/components/form-checkbox-group)
- [Figma](https://www.figma.com/file/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?type=design&node-id=4224-19390&mode=design&t=TXd3ffU1Nuvwi6fI-4)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-02-28 om 12 01 26](https://github.com/nl-design-system/backlog/assets/545207/74dee6b4-98a6-40db-88e5-40f3b14dc546)

Invalid
![Scherm­afbeelding 2024-02-28 om 12 01 41](https://github.com/nl-design-system/backlog/assets/545207/cd3cac5b-c382-4314-a71d-46d1b134ce12)


**Figma**

Checkbox group
![Scherm­afbeelding 2024-02-28 om 12 02 50](https://github.com/nl-design-system/backlog/assets/545207/9d1bddad-3720-45f0-a933-eacfcd067b3f)

States Form field checkbox option
![Scherm­afbeelding 2024-02-28 om 12 02 59](https://github.com/nl-design-system/backlog/assets/545207/34ce8d8c-ecd5-4942-9210-161f9f573930)

---

### Notities

Bekijk ook de [GitHub Discussion - Checkbox toevoeging van RVO](https://github.com/orgs/nl-design-system/discussions/306#discussioncomment-10797851).

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css-form-field-checkbox-group--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/packages/components-react/checkbox-group-react)
- [Figma](https://www.figma.com/file/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?type=design&node-id=954-2227&mode=design&t=iD1JPMmCXMtV7kwf-4)

---

### Screenshots / Varianten

**Storybook**

with only a label
![Scherm­afbeelding 2024-02-28 om 12 08 18](https://github.com/nl-design-system/backlog/assets/545207/818f3b43-c49b-448c-a0e8-01fb8410daee)

with a checkbox group description
![Scherm­afbeelding 2024-02-28 om 12 08 22](https://github.com/nl-design-system/backlog/assets/545207/d6d91ef1-057c-4d87-a3f5-ed00b24e4d07)

with checkbox option descriptions
![Scherm­afbeelding 2024-02-28 om 12 08 28](https://github.com/nl-design-system/backlog/assets/545207/4c2f38c8-e4a5-4090-a54a-df3df4562766)

with checkbox invalid option descriptions
![Scherm­afbeelding 2024-02-28 om 12 08 33](https://github.com/nl-design-system/backlog/assets/545207/54ce2607-c619-4342-a515-2de3cae298a2)

with checkbox invalid group description
![Scherm­afbeelding 2024-02-28 om 12 08 38](https://github.com/nl-design-system/backlog/assets/545207/23f6420a-12fd-41e6-9272-22411e4e6a35)


**Figma**

Checkbox group
![Scherm­afbeelding 2024-02-28 om 12 11 05](https://github.com/nl-design-system/backlog/assets/545207/c55cb969-51c9-4e80-8cca-5503d2032510)

Mogelijkheid voor description per optie.
![Scherm­afbeelding 2024-02-28 om 12 11 21](https://github.com/nl-design-system/backlog/assets/545207/1835c3d3-cdb0-49f7-b8ef-41533bf9cae2)

---

### Notities
- Figma component heeft eenzelfde description opzet als bij [radio group](https://github.com/orgs/nl-design-system/discussions/222#discussioncomment-8607551).
- Figma component heeft eenzelfde validatie opzet als bij [radio group](https://github.com/orgs/nl-design-system/discussions/222#discussioncomment-8607551).
- Bekijk ook de [GitHub Discussion - Checkbox toevoeging van Utrecht](https://github.com/orgs/nl-design-system/discussions/306#discussioncomment-10798361).

**Comment by MariekeBrouwer** [comment]
Bij gemeente Groningen zie je een aantal toepassingen: 
![image](https://github.com/nl-design-system/backlog/assets/149144374/60724800-3ee7-4d56-9a5c-61e6ffccca21)
![image](https://github.com/nl-design-system/backlog/assets/149144374/ee8cacbb-1fd4-434f-bcaa-f88fff63cb9d)
Voorbeelden uit onze Mijn omgeving (testomgeving) en Open Formulieren
- als selectie voor communicatievoorkeuren of voor het invullen van opties voor een formulier (waarbij meerdere antwoorden als 'juist' kunnen worden gezien. 

![image](https://github.com/nl-design-system/backlog/assets/149144374/66d1ae92-679e-4047-92b6-c35fe5d42b24)
- Ook vanuit Open Formulieren, maar dan andere toepassing: bij validatie/verklaring van

We hebben geen inzichten uit onderzoek rondom checkboxes vooralsnog. Maar onze persoon voor toegankelijkheid heeft wel gewezen op voldoende klikvlak ruimte voor de checkboxes. En dat onze kleur rood #E10019 met wit #ffffff wat toegankelijkheidsissues op zou gooien. Niet officieel qua contrast - want daaraan voldoet het, maar dat het ondanks die check wel slecht te lezen is. Schijnt vaker met de kleur rood een issue te zijn. De tekst die wij voor error messages beschikbaar hebben is dan wel erg donker (lijkt veel op zwart), dus wij zijn nog bezig met kijken hoe we die anders kunnen maken.

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Checkbox Group op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=checkbox+group)

**Comment by Marwaxhello** [comment]
## Developer Overheid

[RHC Storybook](https://rijkshuisstijl-community.vercel.app/?path=/docs/rhc-form-field-checkbox-group--docs)

Ik heb op de website van de overheid voor developers een (filterfunctie) `FormfieldCheckboxGroup` gevonden, waarbij naast elk label een nummer staat dat aangeeft hoeveel items er van dat type zijn.

<img width="412" alt="Screenshot 2025-02-10 at 08 56 23" src="https://github.com/user-attachments/assets/7f7a9d4a-d877-42c9-aad6-110a815ddb87" />

Er kan een nummer in de 'beschrijving' worden weergegeven, maar het is geen echte beschrijving. In eerste instantie dacht ik dat het misschien een `Number Badge` zou kunnen zijn, maar dat bleek niet helemaal te kloppen

<img width="412" alt="Screenshot 2025-02-10 at 10 29 58" src="https://github.com/user-attachments/assets/d01a34df-963e-4e62-925b-d445b0294e99" />
